js/luno.js

```diff
@@ -454,6 +475,97 @@ module.exports = class luno extends Exchange {
         return await this.privatePostStoporder (this.extend (request, params));
     }
 
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        await this.loadAccounts ();
+        let currency = undefined;
+        let id = this.safeString (params, 'id'); // account id
+        let min_row = this.safeValue (params, 'min_row');
+        let max_row = this.safeValue (params, 'max_row');
+        if (id === undefined) {
+            if (code === undefined) {
+                throw new ArgumentsRequired (this.id + ' fetchTransactions() requires a currency code argument if no account id specified in params');
+            }
+            currency = this.currency (code);
+            const accountsByCurrencyCode = this.indexBy (this.accounts, 'currency');
+            const account = this.safeValue (accountsByCurrencyCode, code);
+            if (account === undefined) {
+                throw new ExchangeError (this.id + ' fetchTransactions() could not find account id for ' + code);
+            }
+            id = account['id'];
+        }
+        if (min_row === undefined && max_row === undefined) {
+            max_row = 0; // Default to most recent transactions
+            min_row = -1000; // Maximum number of records supported
+        } else if (min_row === undefined || max_row === undefined) {
+            throw new ExchangeError (this.id + " fetchTransactions() require both params 'max_row' and 'min_row' or neither to be defined");
+        }
+        if (limit !== undefined && max_row - min_row > limit) {
+            if (max_row <= 0) {
+                min_row = max_row - limit;
+            } else if (min_row > 0) {
+                max_row = min_row + limit;
+            }
+        }
+        if (max_row - min_row > 1000) {
+            throw new ExchangeError (this.id + " fetchTransactions() requires the params 'max_row' - 'min_row' <= 1000");
+        }
+        const request = {
+            'id': id,
+            'min_row': min_row,
+            'max_row': max_row,
+        };
+        const response = await this.privateGetAccountsIdTransactions (this.extend (params, request));
```

Comment:
Hi!

Can you plz paste some verbose output from this endpoint?
Unfortunately, luno's API docs are not very clear about the purpose of the endpoint, and I think it might be a [`fetchLedger` endpoint](https://github.com/ccxt/ccxt/wiki/Manual#ledger), rather than an endpoint for transactions, since they have a standalone endpoint for withdrawals...

Reply:
Hi.
This is indeed more a  fetchLedger than fetchTransactions. It has a I did not know there was a fetchLedger function. Do you have some documentation on it or can you point me to an exchange with a good implementation? I'll then rewrite this as a fetchLedger instead. 

Below is a sample.
`{'row_index': 130,
   'timestamp': 1593433250517,
   'balance': 0.8000002,
   'available': 0.8000002,
   'account_id': '8535691064011841924',
   'balance_delta': 0.7209352,
   'available_delta': 0.7209352,
   'currency': 'XBT',
   'description': 'Received Bitcoin into Luno',
   'details': {'Address': '38ngV5pkJj6b1m81yXH5cmkWHoFXwLxiVZ',
    'Approximate value': '~ZAR\xa0120,564.15 (167,233 ZAR/BTC)',
    'Transaction': '32f3f96807ea83949ae2641c5a7cba2a6c110ef9202cb7e5d469c2b5c094c305'},
   'detail_fields': {'crypto_details': {'address': '38ngV5pkJj6b1m81yXH5cmkWHoFXwLxiVZ',
     'txid': '32f3f96807ea83949ae2641c5a7cba2a6c110ef9202cb7e5d469c2b5c094c305'}}},
  {'row_index': 129,
   'timestamp': 1593431455031,
   'balance': 0.079065,
   'available': 0.079065,
   'account_id': '8535691064011841924',
   'balance_delta': 0.0,
   'available_delta': 0.0,
   'currency': 'XBT',
   'description': 'Released funds for order BXJKXRU6BGKNJ3Y',
   'details': {}},
  {'row_index': 128,
   'timestamp': 1593431454929,
   'balance': 0.079065,
   'available': 0.079065,
   'account_id': '8535691064011841924',
   'balance_delta': -0.004484,
   'available_delta': 0.0,
   'currency': 'XBT',
   'description': 'Sold 0.004484 BTC/ZAR @ 167,234',
   'details': {}},`

Reply:
@DelAnt the docs are here:

- https://github.com/ccxt/ccxt/wiki/Manual#ledger

And you can find all implementations of it by a simple ctrl+f/cmd+f for `async fetchLedger` throughout the existing derived exchange files, i'd say, Kraken, OKEX, KuCoin or BitMEX would be a good start. Thx again for your involvement!

Reply:
Code has been updated now for fetchLedger instead of fetchTransactions. Let me know what you think or if you want me to make some small changes to the 'type' field.